src/parsemultipart.jl

```diff
@@ -0,0 +1,52 @@
+function parse_multipart_chunk!(chunk, d)
+    re = r"(\n){2,}|(\r){2,}|[\r\n]{4,}"
+    Parsers.exec(re, chunk) || return nothing
+    i, j = re.ovec
+    description = String(chunk[1:i])
+    content = Parsers.nextbytes(re, chunk)
+
+    v = match(r"Content-Disposition: form-data; name=\"(.*)\"; filename=\"(.*)\"[\r\n]+Content-Type: (\S*)", description)
+    if v !== nothing
+        push!(d["name"], v[1])
+        push!(d["filename"], v[2])
+        push!(d["contenttype"], v[3])
+    else
+        v = match(r"Content-Disposition: form-data; name=\"(.*)\"", description)
+        v == nothing && return
+        push!(d["name"], v[1])
+        push!(d["filename"], nothing)
+        push!(d["contenttype"], nothing)
+    end
+
+    re = r"(?:[\r\n]*)(?:-*)$"
+    Parsers.exec(re, content) && (content = content[1:re.ovec[1]])
+    push!(d["content"], content)
+end
+
+parse_multipart_body(body::AbstractArray{UInt8}, boundary) =
+    parse_multipart_body(String(body), boundary)
+
+function parse_multipart_body(body, boundary)
+    body = String(body)
+    dict = Dict(
```

Comment:
I feel like my personal preference would be to return a `Vector{Multipart}` where `Multipart` was a simple struct like:
```julia
struct Multipart
    name::String
    filename::Union{Nothing, String}
    contenttype::Union{Nothing, String}
    content::Vector{UInt8}
end
```
It just seems like it would be a bit simpler on the server side if I could just call this function and iterate over the multparts, dealing w/ each as needed, instead of needing to do something like:
```julia
for i in 1:length(multiparts)
    nm = multiparts["name"][i]
    filename = multiparts["filename"][i]
    # etc....
end
```

Reply:
There already is a `Multipart` struct which I'd like to reuse, except it stores the content as `IO` object, whereas for me a `Array` (actually `SubArray`) of `UInt8` would be better.  Should I simply write my `SubArray` to a `IOBuffer` and put that instead?